### DISCUSSION TITLE
Is there a way to save authStore locally?

### DISCUSSION THREAD

**Discussion by HEAVYPOLY** [pocketbase/pocketbase#3309] [Q&A] [answered]
Is there a way to save pb.authStore.model data locally and retrieve it while offline?
in this example to save favorites (which is a field of user record in pb)
```async function saveFavorites () {
  favorites = []
  for (B of toolBar.children) {
    if (B.tool) favorites.push(B.tool)
  }
  console.log('SAVING FAVORITES', pb.authStore, favorites)
  pb.authStore.model.favorites = favorites
  // localStorage.setItem('favorites', JSON.stringify(favorites))
  // localStorage.setItem('pocketbase_auth', JSON.stringify({ 
  //   token: pb.authStore.token,
  //   model: pb.authStore.model,
  // }))
  await pb.collection('users').update(user.id, { favorites })
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not entirely sure that I understand what do you mean.

If you use the JS SDK `pb.authStore` will keep the auth data in `localStorage` by default (see https://github.com/pocketbase/js-sdk#auth-store).

If you want to modify the existing auth state, you can call `pb.authStore.save(token, model)` with your changed auth data.

**Comment by HEAVYPOLY** [comment]
pb.authStore.save is what I was looking for, works great.  Thank you!
